### PR TITLE
GH-34579: [Python][Docs] TableGroupBy.aggregate options

### LINKED ISSUE #34579: DOC: TableGroupBy.aggregate options
### Describe the bug, including details regarding any error messages, version, and platform.

Context: trying to figure out what it would take to let pandas dispatch to pyarrow's groupby machinery.

Most of the relevant methods we would want to dispatch are reduction-like, which maps onto TableGroupBy.aggregate.  The only docs I see are https://arrow.apache.org/docs/python/generated/pyarrow.TableGroupBy.html#pyarrow.TableGroupBy.aggregate which says `aggregate` takes "List of tuples made of aggregation column names followed by function names and optionally aggregation function option"

It is not clear to me what "function names" are available.  The only example I see is "sum".  

It is also not clear what can be done with FunctionOptions.

### Component(s)

Python

### PR DESCRIPTION
**Pull request by AlenkaF**
### Rationale for this change

Add more information and examples to `pa.TableGroupBy.aggregate` method to make it clearer to use.

### What changes are included in this PR?

Changes in the `pa.TableGroupBy.aggregate` docstrings include:
- link to https://arrow.apache.org/docs/python/compute.html#grouped-aggregations
- extra examples
* Closes: #34579

### CODE REVIEW

File: python/pyarrow/table.pxi [hunk 1]
```diff
@@ -5515,6 +5515,9 @@ list[tuple(str, str, FunctionOptions)]
             column names, for unary, nullary and n-ary aggregation functions
             respectively.
 
+            For the list of function names and respective aggregation
+            function options see: :ref:`py-grouped-aggrs`.
```
Comment (jorisvandenbossche) [inline python/pyarrow/table.pxi:5519]:
```suggestion
            function options see :ref:`py-grouped-aggrs`.
```

File: python/pyarrow/table.pxi [hunk 2]
```diff
@@ -5527,20 +5530,58 @@ list[tuple(str, str, FunctionOptions)]
...
+        values_sum: [[3,7,5]]
+        keys_count: [[2,2,1]]
+        keys: [["a","b","c"]]
+
+        Count the number of non-null values for column "values"
+        over the grouped column "keys":
+
+        >>> import pyarrow.compute as pc
+        >>> t.group_by(["keys"]).aggregate([
+        ...    ("values", "count", pc.CountOptions(mode="all"))
```
Comment (jorisvandenbossche) [inline python/pyarrow/table.pxi:5574]:
```suggestion
        ...    ("values", "count", pc.CountOptions(mode="only_valid"))
```

If you want the "number of non-null values" as mentioned above, you need this option (which is actually the default, but OK to show it explicitly I think)

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good, thanks!

**Comment by AlenkaF** [comment]
@github-actions crossbow submit preview-docs

**Comment by AlenkaF** [comment]
Generated docs for the change in this PR can be viewed here: http://crossbow.voltrondata.com/pr_docs/34759/python/generated/pyarrow.TableGroupBy.html#pyarrow.TableGroupBy.aggregate

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**